packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -244,7 +246,7 @@ If you provided a React element for the optionText prop, you must also provide t
     }, [shouldRenderSuggestions, noOptionsText]);
 
     const getRecordRepresentation = useGetRecordRepresentation(resource);
-    const { getChoiceText, getChoiceValue, getSuggestions } = useSuggestions({
+    const { getChoiceText, getChoiceValue } = useSuggestions({
```

Comment:
we don't need to filter suggestions in AutocompleteInput, because:

- when used in standalone, AutocompleteInput already knows how to filter options
- when used inside a ReferenceInput, it's the responsibility of ReferenceInput to filter

Therefore, `getSuggestions` is useless.

Reply:
Wait, I may have been a bit hasty on this one... I'll investigate a bit more.